### PR TITLE
BE-1396 Changing enroll command to use the new architecture

### PR DESCRIPTION
**Pull request by josecordaz**
**JIRA:** https://pinpt-hq.atlassian.net/browse/BE-1396

**Description:**

- Changing enroll command to use the new architecture

### CODE REVIEW

File: cmd/enroll.go [hunk 1]
```diff
@@ -666,18 +666,26 @@ func runCommand(command string, entity string, opts *commonOptions, integrations
...
 			integrationNames = append(integrationNames, integrationName)
 		}
 
 	} else {
-		authJSON, exclusions, inteName := getAuthInfoFromIntegration(integrations)
-		err := getSetArgs(authJSON, inteName, opts, &args, exclusions)
+		authJSON, exclusions, inteName, err := getAuthInfoFromIntegration(opts, integrations)
+		err = getSetArgs(authJSON, inteName, opts, &args, exclusions)
+		if err != nil {
+			log.Debug(*opts.logger, "error getting integration info", "err", err)
```
Comment (jhaynie) [inline cmd/enroll.go]: move error check up one line

File: cmd/enroll.go [hunk 2]
```diff
@@ -203,6 +203,7 @@ func createValidationRequest(opts *commonOptions) action.Action {
 
 			// onboardSpinner.Successf("%s onboard succeed", integration.Name)
 		} else {
+			panic(valid.Message)
```
Comment (jhaynie) [inline cmd/enroll.go]: probably don't want a panic here

File: Gopkg.toml [hunk 3]
```diff
@@ -122,8 +120,20 @@
...
   name = "github.com/zoonman/gravatar"
   version = "0.0.1"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/pinpt/integration-sdk"
+
+[[constraint]]
+  name = "github.com/pinpt/go-datamodel"
+  version = "0.0.304"
```
Comment (jhaynie) [inline Gopkg.toml]: you cannot use go-datamodel in this project! this is a private module.
you must use integration-sdk only. the integration-sdk only contains public models whereby go-datamodel contains both private and public. the go-datamodel is only visible internally.

File: cmd/enroll.go [hunk 4]
```diff
@@ -166,15 +166,11 @@ func createValidationRequest(opts *commonOptions) action.Action {
 		if valid.Success {
 			validateSpinner.Success(fmt.Sprintf("Integration %s validated", integration.Name))
 			integrationResponse.RefType = integration.Name
+			integrationResponse.RefID = integrationRequest.Integration.ID
+			integrationResponse.IntegrationID = integrationRequest.Integration.ID
```
Comment (rblalock) [inline cmd/enroll.go]: Should we just remove the integration_id from the data model then?

Comment (josecordaz) [reply cmd/enroll.go]: I think we should delete it.

Comment (rblalock) [reply cmd/enroll.go]: https://github.com/pinpt/datamodel/pull/110 done

File: cmd/enroll.go [hunk 5]
```diff
@@ -652,6 +645,10 @@ var enrollCmd = &cobra.Command{
 			}
 			defer exportSubscription.Close()
 
+			spinner.UpdateMessage("Give our listener time to be added")
+			time.Sleep(time.Second * 3) // give our listener time to be added
```
Comment (jhaynie) [inline cmd/enroll.go]: we shouldn't need this ... what is the sleep for?

File: cmd/validate.go [hunk 6]
```diff
@@ -126,7 +126,7 @@ var validateCmd = &cobra.Command{
 					fmt.Fprintf(os.Stderr, `{"success":%v,"message":%s}`, false, strings.TrimSpace(string(buf)))
 				} else {
 					fmt.Fprintf(os.Stderr, `{"success":%v,"status":`, h.failures == 0)
-					fmt.Fprint(os.Stderr, pjson.Stringify(h.statuses))
+					fmt.Fprint(os.Stderr, strings.ReplaceAll(pjson.Stringify(h.statuses), "\"", "\\\""))
```
Comment (jhaynie) [inline cmd/validate.go]: why are you doing this? this is error prone
just make a struct with the data and call pjson.Stringify on that.

Comment (josecordaz) [reply cmd/validate.go]: my bad.

File: cmd/enroll.go (hunk 5 above)
Comment (josecordaz) [reply cmd/enroll.go]: I took it from the unittests, but will remove it.

File: cmd/enroll.go [hunk 7]
```diff
@@ -189,17 +167,21 @@ func createValidationRequest(opts *commonOptions) action.Action {
 			validateSpinner.Success(fmt.Sprintf("Integration %s validated", integration.Name))
 			integrationResponse.RefType = integration.Name
 			integrationResponse.RefID = hash.Values(*opts.agentUUID, integration.Name)
-			integrationResponse.RequestID = ""
+			// integrationResponse.RequestID = "requestID"
```
[the remark this answers is not in the document]
Comment (josecordaz) [reply cmd/enroll.go]: done

File: integrations/bitbucket/onboard.go [hunk 8]
```diff
@@ -42,7 +42,7 @@ func onboardUsers(ctx context.Context, logger log.Logger, bbapi *api.BitbucketAP
 			for _, user := range users {
 				oUser := agent.UserResponseUsers{
 					RefType:    "bitbucket",
-					RefID:      fmt.Sprintf("%d", user.UUID),
+					RefID:      user.UUID,
 					CustomerID: customerID,
```
Comment (jhaynie) [inline integrations/bitbucket/onboard.go]: legit crash here easily fixed in Visual Studio Code or running lint

File: integrations/jira/api/http.go [hunk 9]
```diff
@@ -356,7 +356,15 @@ func doRequest(
...
-		url = pstrings.JoinURL(prefix.(string), "rest", urlPath)
+		if prefix != nil {
+			url = pstrings.JoinURL(prefix.(string), "rest", urlPath)
+		} else {
+			baseurl, err := urlForContext(ctx)
+			if err != nil {
+				return nil, err
+			}
+			url = pstrings.JoinURL(baseurl, urlPath)
+		}
```
Comment (jhaynie) [inline integrations/jira/api/http.go]: crash here but likely related to changes in config handling

File: integrations/jira/export.go [hunk 10]
```diff
@@ -376,7 +376,7 @@ func fetchIssues(ctx context.Context, logger log.Logger, customerID string, dir
 				api.ProcessRawIssueFields(issue, rawIssues, customFieldsFilterRegexp, sprints)
 				err := api.SetIssueURL(ctx, issue)
 				if err != nil {
-					return fmt.Errorf("Error while setting issue's URL", err)
+					return fmt.Errorf("Error while setting issue's URL: %v", err)
 				}
```
Comment (jhaynie) [inline integrations/jira/export.go]: legit crash here easily fixed in Visual Studio Code or running lint

File: pkg/upload/uploader/uploader_test.go [hunk 11]
```diff
@@ -86,6 +87,6 @@ func TestSetLogger(t *testing.T) {
 func TestInternalLoggerDebugPrints(t *testing.T) {
 	SetLogger(nil, ",", 10, false)
 	il := new(internalLogger)
-	il.debugPrintf("asd", nil)
+	il.debugPrintf("asd %v", nil)
 	il.debugPrintln("asd", nil)
```
Comment (jhaynie) [inline pkg/upload/uploader/uploader_test.go]: legit crash here easily fixed in Visual Studio Code or running lint

**Pull request merged**